test/integration/fetch-collection/index.js

```diff
@@ -1,117 +1,138 @@
-                expect(res.body.data).to.be.an("array");
-              });
-
-              // "Unless otherwise noted, objects defined by this
-              //  specification MUST NOT contain any additional members."
-              it("should not contain extra members", () => {
-                const isAllowedKey = (key) =>
-                  ["type", "id", "attributes", "relationships", "links", "meta"].indexOf(key) !== -1;
-
-                if(!Object.keys(res.body.data[0]).every(isAllowedKey)) {
-                  throw new Error("Invalid Key!");
-                }
-              });
-
-              it("should contain links under each relationship", () => {
-                let liaisonRelationships = res.body.data.map(it => it.relationships.liaisons);
-                liaisonRelationships.forEach((it) => {
-                  // Every liaison should have a links object,
-                  // and self and related must not be outside of links.
-                  expect(it.links).to.be.an("object");
-                  expect(it.self).to.be.undefined;
-                  expect(it.related).to.be.undefined;
-
-                  expect(it.links.self).to.be.a("string");
-                  expect(it.data).to.not.be.undefined; //can be null, though
-                });
-              });
-            });
-          });
-        });
-      }).done();
+describe("Fetching Collection", () => {
+  let res, Agent;
 
-    Agent.request("GET", "/people?sort=gender")
-      .accept("application/vnd.api+json")
-      .promise()
-      .then((res) => {
-        describe("Fetching Ascending Gendered Collection", () => {
-          it("should have Jane above John", () => {
-            let johnJaneList = res.body.data.map((it) => it.attributes.name).filter((it) => {
-              return ["John Smith", "Jane Doe"].indexOf(it) > -1
-            });
-            expect(johnJaneList[0]).to.equal("Jane Doe");
-            expect(johnJaneList[1]).to.equal("John Smith");
-          });
-        })
-      }).done();
-
-    Agent.request("GET", "/people?sort=-name")
-      .accept("application/vnd.api+json")
-      .promise()
-      .then((res) => {
-        describe("Fetching Descended Sorted Name Collection", () => {
-          it("Should have John above Jane", () => {
-            let johnJaneList = res.body.data.map((it) => it.attributes.name).filter((it) => {
-              return ["John", "Jane"].indexOf(it.substring(0, 4)) > -1
-            });
-            expect(johnJaneList[0]).to.equal("John Smith");
-            expect(johnJaneList[1]).to.equal("Jane Doe");
-          });
+  before(done => {
+    AgentPromise.then(A => {
+      Agent = A;
+      return Agent.request("GET", "/organizations")
+        .accept("application/vnd.api+json")
+        .promise();
+    }, done).then(response => {
+      res = response;
+      done();
+    }, done).catch(done);
+  });
+
+  describe("Status Code", () => {
+    it("should be 200", (done) => {
```

Comment:
A bunch of these tests, for now reason, used to be async (i.e. take and call a `done` argument). I fixed that [here](https://github.com/ethanresnick/json-api/commit/c8985cae961ba44ed3fdef686dd42f658813be7d#diff-4604d069ddcba9c10f67be20af484e45L12), but it looks like that change got lost amidst the merges. Can you remove the `done`s again from the tests here that don't need them?